src/frontend/apps/desk/src/features/mail-domains/components/MailDomainsContent.tsx

```diff
@@ -1,51 +1,101 @@
-import { DataGrid } from '@openfun/cunningham-react';
+import { DataGrid, Loader, usePagination } from '@openfun/cunningham-react';
+import React, { useEffect } from 'react';
 import { useTranslation } from 'react-i18next';
 
-import { Card } from '@/components';
+import { Box, Card, Text, TextErrors } from '@/components';
+import { MailDomain } from '@/features/mail-domains';
+import { useMailDomain } from '@/features/mail-domains/api/useMailDomain';
+import { useMailDomainMailboxes } from '@/features/mail-domains/api/useMailDomainMailboxes';
+import { PAGE_SIZE } from '@/features/mail-domains/conf';
 
-export function MailDomainsContent() {
+import { default as MailDomainsLogo } from '../assets/mail-domains-logo.svg';
+export type ViewMailbox = { email: string; id: string };
+
+export function MailDomainsContent({ mailDomain }: { mailDomain: MailDomain }) {
   const { t } = useTranslation();
+  const pagination = usePagination({
+    defaultPage: 1,
+    pageSize: PAGE_SIZE,
+  });
+  const { page, pageSize, setPagesCount } = pagination;
 
-  const dataset = [
-    {
-      id: '1',
-      name: 'John Doe',
-      email: 'john@doe.com',
-      state: 'Active',
-      lastConnection: '2021-09-01',
-    },
-    {
-      id: '2',
-      name: 'Jane Doe',
-      email: 'jane@doe.com',
-      state: 'Inactive',
-      lastConnection: '2021-09-02',
-    },
-  ];
+  const { data, isLoading, error } = useMailDomainMailboxes({
+    id: mailDomain.id,
+    page,
+  });
 
-  return (
-    <Card $padding="small" $margin="large">
-      <DataGrid
-        columns={[
-          {
-            headerName: t('Names'),
-            field: 'name',
-          },
-          {
-            field: 'email',
-            headerName: t('Emails'),
-          },
-          {
-            field: 'state',
-            headerName: t('State'),
-          },
-          {
-            field: 'lastConnection',
-            headerName: t('Last Connecttion'),
-          },
-        ]}
-        rows={dataset}
-      />
-    </Card>
+  const viewMailboxes: ViewMailbox[] =
+    mailDomain && data?.results?.length
+      ? data.results.map((mailbox) => ({
+          email: `${mailbox.local_part}@${mailDomain.name}`,
```

Comment:
I can see there is the data `secondary_email`, is it not interesting to display this data in the datagrid?

Reply:
If I remember correctly, with @sdemagny we thought not to show the secondary email address: we consider it information only the mailbox owner should know.